src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/ContigAlignmentsModifier.java

```diff
@@ -208,18 +208,46 @@ private static AlignmentInterval clipAlignmentInterval(final AlignmentInterval i
                 refBasesConsumed += ce.getLength();
             }
         }
+        if (newMiddleSection.size() < 2)
+            throw new GATKException("After clipping, the middle section of an alignment isn't split, which seems wrong\t" + input.toPackedString() + "\tclip length " +
+                    clipLengthOnRead + "\tclip from end " + (clipFrom3PrimeEnd? "3":"5"));
+
+        // guard against edge case where the requested clipping clip away a whole alignment block that neighbors an indel operation
+        // e.g. "30M5I..." and user requests 30 bases to be chipped away hence leading to "...S5I", which we consider to be invalid
+        final CigarElement firstOp = newMiddleSection.get(0);
+        final CigarElement secondOp = newMiddleSection.get(1);
+        if ( firstOp.getOperator().equals(CigarOperator.S) && secondOp.getOperator().isIndel() ) {
+            // the input cigar, if good, cannot have neighboring INS/DEL operations so we are safe here
+            if (newMiddleSection.size() < 3)
+                throw new GATKException("After clipping, the middle section of an alignment contains an indel operation neighboring the new clipping but nothing else, " +
+                        "which seems wrong\t" + input.toPackedString() + "\tclip length " +
+                        clipLengthOnRead + "\tclip from end " + (clipFrom3PrimeEnd? "3":"5"));
+
+            refBasesConsumed += secondOp.getLength();
+            if (secondOp.getOperator().equals(CigarOperator.D)) {
```

Comment:
would it be more clear to say `if (! secondOp.consumesReadBases())` ? or add a quick comment explaining that that's why you're doing this?

Reply:
Sorry for the confusion.
Up in the very beginning of this function we guarded against padding or `N` bases, so we are down to alignment, clipping, or gap from then onward. Then in the enclosing `if` block, we assumed the second operation is an `isIndel()`.
Do that make sense?

Reply:
Yes, it does, and I understood that.. I was just suggesting that using `consumesReadBases` might actually make it clearer to the reader why you are just ditching the cigar element in one case vs adding it to the clipping in the other. But it's definitely no big deal, don't worry about changing it.